### ISSUE TITLE
Bug - Blank screen when I click on Server Side Rendering suggestion

### ISSUE THREAD

**Issue by RayMathew** [Buzzpy/Dev-Encyclopedia#82]
It's because the suggestion is spelled "Server Side Rendering (SSR)" while the card title is spelled "Server-side rendering".

There might be similar issues with other cards too.

@Rodelph would it be because of your autocomplete commits?

**Comment by Rodelph** [comment]
That is indeed because of the autocomplete. The way the autocomplete works is that there is a function that searches through the content/terms/*.json files and extracts all the names from there and load it into an array. The thing is that the files sometimes contain "_" which serves as a space but i didn't have a check for other special character such as '-'. I believe i should change the method of feting the names and that should fix this bug. 

I will look at it when i will be at home tonight.

Thanks for the info.

**Comment by Rodelph** [comment]
Coming back to you on the bug. I have fixed the code where now i load an array by reading the titles of the json files instead of the file name itself and it works.

![Recording 2024-08-20 at 18 06 29](https://github.com/user-attachments/assets/f5d4a182-dd05-4419-b8e8-53b90d7c9569)

Since the merger to main (new master) i have noticed that when we search all the buttons become invisible. Is this another bug or you guys made it a feature ? You can only see the buttons when nothing is being searched.

**Comment by Buzzpy** [comment]
Great, thanks a bunch!

And that "button-disappearance" thing was a feature, @Rodelph  😄
You see, on desktop it's fine but on mobile, even after the card for the searched term is shown, users have to scroll down to see the card.

Is it a bad feature? I hope not.

**Comment by RayMathew** [comment]
> Is it a bad feature? I hope not.

It isn't intuitive. Let's say that I search for some card, then I think "Hmm I really like this website, how do I contribute? Oh right, I have to clear the search first!"

Doesn't make sense right? 

Yeah the mobile UI is maybe a problem, but we need a better solution for it. Even I immediately just thought of a menu page. Let's start a discussion first to brainstorm ideas with everyone @Buzzpy ?
@Rodelph implementation details with astro we can always just figure it out :)

**Comment by Rodelph** [comment]
Ye i agree with you, it's not that intuitive. People will think that the website it broken on mobile and honestly even web. The user should always have the choice to navigate through other things via those buttons other wise it becomes clunky. Imo we should have a css for mobile that should take care of that and keep the buttons for PCs. 

Also i was working on the develop branch and this is a problem too that wasn't in the previous patch (not even on production). The suggestions are not showing up entirely (covered by cards).

![image](https://github.com/user-attachments/assets/cd69e5b3-71ee-4fc9-a3be-e39654e5451f)

The bug is easy to fix but i think we should fix the UI first just for PCs at least.

**Comment by RayMathew** [comment]
I'll look into the focus issue 👍🏻

**Comment by RayMathew** [comment]
Btw, if you fixed the blank screen issue this Issue 82 should be closed, and you folks can add the details of the hidden buttons problem elsewhere. @Rodelph @Buzzpy

**Comment by Rodelph** [comment]
Once the PR is merged, i will ask @Buzzpy to close it.

**Comment by Buzzpy** [comment]
> Yeah the mobile UI is maybe a problem, but we need a better solution for it. Even I immediately just thought of a menu page. Let's start a discussion first to brainstorm ideas with everyone @Buzzpy ? @Rodelph implementation details with astro we can always just figure it out :)

Thanks for the views, @RayMathew and @Rodelph ! Yup, I think you're right. Can you start a discussion on Discussions, please? I think you can explain the issue better than me :D

> Also i was working on the develop branch and this is a problem too that wasn't in the previous patch (not even on production). The suggestions are not showing up entirely (covered by cards). 

I think it's because of my "button-disappearance" thing. I'm middle of an interview task at the moment, I'll check back asap (there's a high possibility it's not today tho).

Thanks a ton again, I'm really grateful to have you two as contributors! Also, please excuse me for any delays in merging or replying, it's a really busy day.

**Comment by Rodelph** [comment]
Ok. I have looked into the history of merges and the commit "Overlap issue fixed" is what made the buttons go away when searching on PC. I have reverted from the change (only the hero-astro.js file was changed to fix this UI issue) and now it's fine.

![Recording 2024-08-20 at 19 34 23](https://github.com/user-attachments/assets/70273203-64e6-431f-9a51-eb8236c3903c)

The change that was made was related to this block that changed the UI on the input of the user. For mobile we will find another way of doing this in another issue.

![image](https://github.com/user-attachments/assets/c288d5ba-69bd-4871-81ca-35acd6316254)

As for the bug that concerns this issue I have made the change so that instead of loading an array of names from the files, we actually read into the files and get the titles. The bug should never happen again unless the title was not specified in the json file. Nevertheless, I added a fallback option so that the we get filename in case of an error such as forgetting to have the title in the json itself (so that the website does not break).

Maybe we should automate the way json files are being generated, and even have a way to test them. I will open an issue for this too. 

The gif above is the changes that i have made with the fixes to UI. @Buzzpy I will open a PR so that you can merge it if possible, and then you can close this issue.

**Comment by Rodelph** [comment]
@Buzzpy you can close this issue. 😉
